challenge.rb

```diff
@@ -1,26 +1,32 @@
 
 def capitalize_each_string(input)
   #implement your solution here
+	return input.map { |string| string.capitalize }
 end
 
 def fetch_the_dog(input)
   #implement your solution here
+	return input.select { |string| string.eql? "dog" }
 end
 
 def no_dogs_allowed(input)
   #implement your solution here
+	return input.reject { |string| string.eql? "dog" }
 end
 
 def count_the_animals(input)
   #implement your solution here
+	return input.count
 end
 
 def fetch_the_first_two(input)
   #implement your solution here
+	return input.take(2)
 end
 
 def fetch_CD_animals(input)
   #implement your solution here
+	return input.find_all { |string| string[0] == 'c' || string[0] == 'd' }
```

Comment:
This works robustly, but is pretty verbose. If you're up for using regular expressions, check out Enumerable#grep. [Link](http://ruby-doc.org/core-2.2.3/Enumerable.html).

Otherwise, I think there are still some ways to make this better. E.g. Enumerable#include?